src/rpc/rawtransaction.cpp

```diff
@@ -1708,12 +1708,19 @@ UniValue converttopsbt(const JSONRPCRequest& request)
 
 UniValue utxoupdatepsbt(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() != 1) {
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
         throw std::runtime_error(
             RPCHelpMan{"utxoupdatepsbt",
-            "\nUpdates a PSBT with witness UTXOs retrieved from the UTXO set or the mempool.\n",
+            "\nUpdates a PSBT with data from descriptors and the UTXO set or the mempool (segwit outputs only)\n",
             {
-                {"psbt", RPCArg::Type::STR, RPCArg::Optional::NO, "A base64 string of a PSBT"}
+                {"psbt", RPCArg::Type::STR, RPCArg::Optional::NO, "A base64 string of a PSBT"},
+                {"descriptors", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG, "An array of either strings or objects", {
+                    {"", RPCArg::Type::STR, RPCArg::Optional::OMITTED, "A descriptor"},
+                    {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "An object with a descriptor and extra information", {
+                         {"desc", RPCArg::Type::STR, RPCArg::Optional::NO, "A descriptor"},
+                         {"range", RPCArg::Type::NUM, "1000", "Up to what index HD chains should be explored"},
```

Comment:
Instead of a single range value, this could be the start and end values of the range like what `importmulti` does?

Reply:
I'm copying the behavior of `scantxoutset` here. Maybe it makes sense to merge the two afterwards, and have all three (scantxoutset, utxoupdatepsbt, importmulti) support both start+end and range?

Reply:
This is now fixed by using the same parameter style as all through #15497.